surfsense_backend/app/agents/researcher/nodes.py

```
@@ -1238,3 +1247,166 @@ async def handle_qna_workflow(state: State, config: RunnableConfig, writer: Stre
+            chat_history_xml += f"<message>{str(message)}</message>\n"
+    chat_history_xml += "</chat_history>"
+    
+    # Format available documents for the prompt
+    documents_xml = "<documents>\n"
+    for i, doc in enumerate(reranked_documents):
+        document_info = doc.get("document", {})
+        source_id = document_info.get("id", f"doc_{i}")
+        source_type = document_info.get("document_type", "UNKNOWN")
+        content = doc.get("content", "")
+        
+        documents_xml += f"<document>\n"
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)